docs/containers/tutorial-django-push-to-registry.md

```diff
@@ -0,0 +1,64 @@
+
+To create an Azure Container Registry:
+
+1. Follow [Quickstart: Create a container registry using the Azure portal](https://docs.microsoft.com/azure/container-registry/container-registry-get-started-portal), but stop before "Push image to ACR". Django deployments must include the settings described below to work properly.
+
+2. Make sure that the registry endpoint you created is visible under **Registries** in the **Docker** explorer of VS Code:
+
+    ![Docker explorer in VS Code showing registries](images/quickstarts/python-django-registries.png)
+
+## Settings for Django apps
+
+1. In your Django project's `settings.py` file, modify the `ALLOWED_HOSTS` list to include the root URL to which you intend to deploy the app. For example, the following code assumes deployment to an Azure App Service (azurewebsites.net) named "vsdocs-django-sample-container":
+
+    ```python
+    ALLOWED_HOSTS = [
+        # Example host name only; customize to your specific host
+        "vsdocs-django-sample-container.azurewebsites.net"
+    ]
+    ```
+
+    Without this entry, you'll see a "DisallowedHost" message after deployment that instructs to you add the website domain to `ALLOWED_HOSTS`. This will require you to rebuild, push, and redeploy the image once again.
+
+2. On the **Command Palette** (`kb(workbench.action.showCommands)`), select **Docker: Build Image** to rebuild image with new settings.
+
+>**Tip**: If you want to test your image in production on multiple hosting services, you can simply input `"*"` in ALLOWED_HOSTS.
+
+## Push the image to a registry
+
+Once `ALLOWED_HOSTS` have been declared, the next step is to push your Django image to a container registry:
+
+1. Ensure you image is tagged with the correct path registry path (e.g `username.azurecr.io/django-tutorial:latest`)
+
+2. Open the **Command Palette** (`kb(workbench.action.showCommands)`) and select **Docker: Push**.
+
+3. Choose the image you just built to push into the registry. Upload progress will appear in the Terminal.
+
+4. Once completed, expand the **Registries** > **Azure** (or **DockerHub**) node in the **Docker** explorer, then expand the registry and image name to see the exact image. (You may need to refresh the **Docker** explorer.)
+
+    ![The built app image in the Azure Container Registry](images/quickstarts/python-django-image-in-acr.png)
+
+> **Tip:** The first time you push an image, you see that VS Code uploads all of the different layers that make up the image. Subsequent push operations, however, upload only the layers that have changed. Because it's typically only your app code that's changes, these uploads happen much more quickly, making for a tight inner loop. To see this, make a small change to your code, rebuild the image, and then push again to the registry. The whole process typically completes in a matter of seconds.
```

Comment:
```suggestion
> **Tip:** The first time you push an image, you see that VS Code uploads all of the different layers that make up the image. Subsequent push operations, however, upload only the layers that have changed. Because it's typically only your app code that changes, these uploads happen much more quickly, making for a tight inner loop. To see this, make a small change to your code, rebuild the image, and then push again to the registry. The whole process typically completes in a matter of seconds.
```

Reply:
The "completes in a matter of seconds" part is machine-dependent and might not be applicable to uses on some older hardware, or it might be shorter in possibly future faster hardware.